### ISSUE TITLE
Error Message in antsMultivariateTemplateConstruction2.sh

### ISSUE THREAD

**Issue by eal09** [stnava/ANTs#283]
Hello,

I have successfully run antsMultivariateTemplateConstruction.sh several times, but I consistently get errors when attempting to run antsMultivariateTemplateConstruction2.sh. It appears the error is occurring at line 266: "imagelist=(`ls ${outputname}template${whichtemplate}*WarpedToTemplate.nii.gz`)". It seems that when this line is first run, the only file which has been created in my output directory is "${outputname}template0.nii", so it throws an error when the specified file above is not found.

Previously, I also received an error at line 218, stating that "AverageImages: command could not be found." Originally, the script read "AverageImages $dim $output 0 ${images[*]}" at line 218. After replacing this with ANTSPATH appended, i.e., "${ANTSPATH}/AverageImages $dim $output 0 ${images[*]}", the script seemed to run normally past this point.

Here is the Cygwin output I receive as the first error occurs: 
![image](https://cloud.githubusercontent.com/assets/17534198/13382150/e820e2ee-de32-11e5-979b-ede38931609c.png)

Thanks,
Elliot

**Comment by ntustison** [comment]
Regarding the second issue---usually ANTSPATH is specified in the PATH, which is why this bug was missed.  However, it's fixed now.  

I'm not sure about the first issue.  Make sure you're using the latest ANTs from the repo and let me know if you can reproduce the error on this  2-D [example](https://github.com/ntustison/TemplateBuildingExample), specifically, the brain slices example using this [script](https://github.com/ntustison/TemplateBuildingExample/blob/master/BrainSlices/templateCommandMultivariateBSplineSyN.sh).

**Comment by ntustison** [comment]
I'm pretty sure this is fixed.  Might need to reopen if the issue still persists.

**Comment by eal09** [comment]
Apologies for the delayed response. It looks like I receive a similar error message when running this on the template building example. I used the following command "
${ANTSPATH}/antsMultivariateTemplateConstruction2.sh -d 2 -o T1_ -i 4 -g 0.25 -j 2 -c 0 -k 2 -w 1x1 -f 8x4x2x1 -s 3x2x1x0 -q 100x70x50x10 -n 1 -r 1 -l 1 -m CC[2] -t TimeVaryingVelocityField[0.25,3,3,0,0,0] OASIS*.nii.gz" from the TemplateMultivariateBSplineSyN.sh script. I've successfully completed the 2D brain slices example before using the command in templateCommandSyN.sh however, which calls antsMultivariateTemplateConstruction.sh instead of antsMultivariateTemplateConstruction2.sh. 

Below is the error message I received this time: 
![image](https://cloud.githubusercontent.com/assets/17534198/13473745/53fce766-e080-11e5-8ddd-0180863b372d.png)

It looks like my directory where the brain slices are now has "job_0_0" files with "metriclog" files, as well as "T1_template0OASIS-TRT-20-16Slice121Repaired.nii" files, "T1_template1OASIS-TRT-20-10Slice121LaplacianRepaired.nii" files, and a "T1_template1.nii". However, there don't appear to be any files which say "...Warp.nii.gz", "...GenericAffine.mat", or "WarpedToTemplate.nii.gz", as the error message states. Also, it looks like the N4Bias correction step ran successfully, albeit with some error message periodically (see below).
![image](https://cloud.githubusercontent.com/assets/17534198/13473881/03bf835c-e081-11e5-9562-6a4326e7c05f.png)

Finally, if it makes a difference, I am running ANTs on windows. 

Thanks,
Elliot

**Comment by ntustison** [comment]
How recent is your ANTs build?  Are you using the latest version along with the latest scripts?

**Comment by eal09** [comment]
Yes, it looks like it may possibly be a version compatibility issue. I initially downloaded and compiled the most recent release from source code using mingGW, CMake, ITK4, and zlib. Although it seemed to work fine for some applications, it may not have compiled completely correctly, so I tried using the pre-compiled Windows version on Sourceforge. BTW, I noticed that Sourceforge currently has a very old version set as the default download for Windows users (it says "Looking for the latest version? Download ANTs-1.9.x-win32.exe (48.0 MB)" from June 2011. This download is getting a fair amount of downloads per week currently based on the site's statistics, and this is what I initially downloaded by mistake. I found it had very limited functions compared to recent releases, both in terms of the C++ applications and shell scripts, and I'm not sure if it was really functional anymore. I fear that a lot of unaware Windows downloaders may be wasting their time on this version. Are there any plans in the near term to update the pre-compiled Windows to version 2.1?

Afterwards, I found and downloaded the most recent Windows version, ANTs-1.9.v4-win32.exe, from Dec. 2012, and updated this with the latest shell scripts and other files. This seems to have mostly full functionality, and recently I've been able to run antsMVTC.2 using both SyN and TimeVaryingVelocityField for transformation methods. However, it still throws the same error for B-Spline SyN. Is this likely due to version incompatibility? antsMVTC.sh seems fully functional. (Note: when running these examples, I noticed that the scripts in the brain slice template building example have their transformation methods mismatched: the templateCommandMultivariateBSplineSyN.sh script is currently set to use TimeVaryingVelocityField[0.25,3,3,0,0,0] and the templateCommandTimeVarying.sh script is set to use BSplineSyN[0.1,26,0].)

Thanks,
Elliot
